### PR TITLE
Throw possible errors from `orderSequentially` and close the container even when the runtime does not own flushing

### PR DESCRIPTION
**Pull request by andre4i**
Affects https://github.com/microsoft/FluidFramework/issues/7365 and https://github.com/microsoft/FluidFramework/issues/8437

### CODE REVIEW

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 1]
```diff
@@ -1660,20 +1660,22 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
...
+        this.setFlushMode(savedFlushMode);
     }
 
     private trackOrderSequentiallyCalls(callback: () => void): void {
         try {
             this._orderSequentiallyCalls++;
             callback();
+        } catch (error) {
+            if (this._orderSequentiallyCalls === 1) {
+                // In case of nesting, only the top level call should close the container
```
Comment (vladsud) [inline packages/runtime/container-runtime/src/containerRuntime.ts:1674]: I'm not sure I agree with this behavior.
I see two options:
1. exceptions go through callstack to most outer batch. What we do here for inner batches does not matter - container gets closed.
2. Exception is eaten somewhere on the way and does not reach outer batch.

What is the recovery path calling code can make in # 2? I think I do not see any. And if there is none, then we end up with partial batch being committed to model and broadcast to the world.

Comment (andre4i) [reply packages/runtime/container-runtime/src/containerRuntime.ts:1674]: I agree. I will rework this and add a test. If a customer calls orderSequentially recursively and handles/swallows the exception, we will miss the error and not close the container.

**Review by vladsud — APPROVED** [review]

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 2]
```diff
@@ -1660,20 +1660,18 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
...
+        this.flush();
+        this.setFlushMode(savedFlushMode);
     }
 
     private trackOrderSequentiallyCalls(callback: () => void): void {
         try {
             this._orderSequentiallyCalls++;
             callback();
+        } catch (error) {
+            this.closeFn(CreateProcessingError(error, "orderSequentially"));
```
Comment (anthony-murphy) [inline packages/runtime/container-runtime/src/containerRuntime.ts:1673]: this isn't really a processing error. processing errors are really more about the processing of incoming state, like ops or snapshot data. not sure what the "right" error would be here

Comment (andre4i) [reply packages/runtime/container-runtime/src/containerRuntime.ts:1673]: My thinking is that it's an error in 'processing' a batch. Let me know if you have strong opinions on this. I was split between this one and `UsageError` but I didn't like the latter as it's not like the client would intentionally misuse this API by throwing exceptions...

**Review by anthony-murphy — APPROVED** [review]

**Pull request merged**